### PR TITLE
MRG: Fix stats examples and wording

### LINKED ISSUE #5636: DOC Be more accurate for 1-sample Monte Carlo-sampling based tests
We must be careful regarding how we document the one-sample permutation test. What exactly is the null hypothesis?

After some comments from Eric Maris and discussion with @larsoner , it seems we should be really careful so not to lead people into writing false things in their methods sections.

See e.g.

https://mne-tools.github.io/dev/generated/mne.stats.permutation_cluster_1samp_test.html#mne.stats.permutation_cluster_1samp_test

vs.

https://mailman.science.ru.nl/pipermail/fieldtrip/2018-August/012309.html

### PR DESCRIPTION
**Pull request by larsoner**
I've fixed plotting in three stats examples with weird percentiles. Eventually I'll fix the wording pointed out by @jona-sassenhagen 

- [x] fix example colorbars (our default percentiles are problematic in these corner-case examples)
- [x] fix wording

Closes #5636.

### CODE REVIEW

**Comment by larsoner** [comment]
`master:`

http://mne-tools.github.io/dev/auto_tutorials/plot_stats_cluster_spatio_temporal.html#visualize-the-clusters

This PR:

https://9888-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_stats_cluster_spatio_temporal.html#visualize-the-clusters

**Comment by jona-sassenhagen** [comment]
I don't understand the colorbar in the example. Why is it so weirdly discontinuous?

The example reads:

>conditions across subjects (simulated here using one subject's data).

This seems unnecessarily complicated. Why not just say, `we look for significant differences across trials (although you can also test across subjects like this)`?

**Comment by mmagnuski** [comment]
Yes, the colorbars in the examples (and sometimes using pysurfer locally) were puzzling me for some time:
image
My understainding is that it is fixed/being fixed in pysurfer but mne still uses the last releast version?

**Comment by larsoner** [comment]
It's just a rendering bug on CircleCI, locally it should look correct

**Comment by mmagnuski** [comment]
Ok, good to know. Locally I had strange cbar limits when using RdBu_r (-1 to 1 IIRC although data limits are paassed to clim were different).

**Comment by larsoner** [comment]
Feel free to open a separate issue for that

**Comment by larsoner** [comment]
>  Why not just say, we look for significant differences across trials (although you can also test across subjects like this)?

Because the typical use case (and one that Maris would be happy about) is to contrast two conditions across a set of subjects

**Comment by larsoner** [comment]
Okay @jona-sassenhagen I think I fixed the wordings, can you have a look to see if this would have saved you some pain?

**Comment by larsoner** [comment]
Okay to merge? I was wondering why my fixes for doc building weren't there when I was building #5514

File: tutorials/plot_background_statistics.py [hunk 1]
```diff
@@ -181,29 +184,36 @@ def plot_t_p(t, p, title, mcc, axes=None):
...
-#           (which is :math:`2^{N_{samp}}-1` for a one-tailed and
-#           :math:`2^{N_{samp}-1}-1` for a two-tailed test, not counting the
-#           veridical distribution), instead of randomly exchanging conditions
-#           the null is formed from using all possible exchanges. This is known
-#           as a permutation test (or exact test).
+# .. warning:: In the case of a true one-sample t-test, i.e. analyzing a single
+#              condition rather than the difference between two conditions,
+#              it is not clear where/how exchangeability applies; see
+#              `this FieldTrip discussion <ft_exch_>`_   # noqa: E501
+#              for more insight. Thus for these non-parametric tests, we will
```
Comment (agramfort) [inline tutorials/plot_background_statistics.py]: sentence not finished

**Comment by larsoner** [comment]
Fixed

**Comment by larsoner** [comment]
https://10067-1301584-gh.circle-artifacts.com/0/html/auto_tutorials/plot_background_statistics.html

**Comment by larsoner** [comment]
> Although maybe someone should check if the sign flip test actually works when the null hypothesis is not exchangeability of two samples.

I think Maris's point (and what I tried to have a `.. warning` about here) is that it very well might not be valid. But in any case, actually doing it is out of scope of this PR

**Comment by agramfort** [comment]
@jona-sassenhagen please merge if you think it's already a progress. We can do better in the next PRs.

**Comment by jona-sassenhagen** [comment]
Yes, I will when it’s green.

**Comment by jona-sassenhagen** [comment]
Great, thanks Eric.

**Pull request merged**